sdk/src/transaction/error.rs

```diff
@@ -161,6 +161,10 @@ pub enum TransactionError {
     /// Sanitized transaction differed before/after feature activiation. Needs to be resanitized.
     #[error("ResanitizationNeeded")]
     ResanitizationNeeded,
+
+    /// Program execution is temporarily restricted.
+    #[error("Program execution is temporarily restricted.")]
+    ProgramExecutionTemporarilyRestricted,
```

Comment:
Could an `account_index` be snuck in here? I'm thinking of cases where a transaction may reference multiple programs, and providing the index of the restricted program could be helpful for debugging or providing an improved error message
```suggestion
    ProgramExecutionTemporarilyRestricted { account_index: u8 },
```

Reply:
assuming there's an `InvokeContext` nearby, we should also log the restricted program's address and a finer grained reason when returning this error

Reply:
> No. The current implementation fails the transaction at accounts load stage. It is before we actually created the InvokeContext. Failing early will reduce the execution time.

in this case, sure. could be others where that's not the case.  i think i'm actually requesting a comment to suggest tx-logging when practical :thinking: 

for this case though, we _should_ ensure that transactions that fail are charged fees. the cluster should not be forced to absorb naive client code

Reply:
@HaoranYi why did I think we added the account index. Did I lose it somehow?

Reply:
yeah. we did.
I think I have it locally.

Reply:
pushed.